census/ui_app/QuestionForm.jsx

```diff
@@ -1,11 +1,96 @@
 import React from 'react';
 import _ from 'lodash';
 
-const QuestionField = React.createClass({
+const QuestionInstructions = React.createClass({
+  render() {
+    if (this.props.instructionText) {
+      return (
+      <div className="instructions">
+        <div className="collapse" id={'instructions' + this.props.id}>
+          <h4>Instructions</h4>
+          <span dangerouslySetInnerHTML={{__html: this.props.instructionText}} />
+        </div>
+        <a className="toggle"
+           role="button"
+           data-toggle="collapse"
+           href={'#instructions' + this.props.id}
+           aria-expanded="false"
+           aria-controls={'instructions' + this.props.id}>
+            <span className="sr-only">Help</span><span className="icon">?</span>
+        </a>
+      </div>
+      );
+    }
+    return (<div className="instructions"></div>);
+  }
+});
+
+const QuestionComments = React.createClass({
+  render() {
+    return (<div className="comments">
+      <label htmlFor={this.props.id + '_comment'}>Comments</label>
+      <textarea placeholder={this.props.placeholder || 'Add comments' }
+                id={this.props.id + '_comment'}
+                rows="5"></textarea>
+    </div>);
+  }
+});
+
+// A base Higher-Order Component providing common behaviour for all Question
+// Fields.
+const baseQuestionField = function(QuestionField) {
+  const BaseQuestionField = React.createClass({
+    _isSub() {
+      return (this.props.position % 1 !== 0);
```

Comment:
Might be clearer if you know more about the possible values, but yeah, a comment wouldn't hurt.